sched/scheduler/task_runner.go

```diff
@@ -35,6 +35,8 @@ type taskRunner struct {
 	taskId string
 	task   sched.TaskDefinition
 	nodeId cluster.NodeId
+
+	abortCh chan interface{}
```

Comment:
Actually, let's make this a "chan bool" to support my use case of node preemption. The meaning will be endSaga=true/false, you'll want to set it to true, and I'll want to set it to false.